### PR TITLE
[feature] Applying Hud

### LINKED ISSUE #60: Add active item HUD

### PR DESCRIPTION
**Pull request by naufalhilmi12345**
### Main changes

- Applying Hud for item but still using sprite logo

### Tasks List

- [ close #60 ]

### Attachments

https://user-images.githubusercontent.com/44135006/150634540-29fbde09-ecce-4e32-9aa5-60f4f74d723e.mp4

### CODE REVIEW

File: Assets/Astralization/Scripts/Hud/Hud.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    public void ActivateHud(Sprite logo)
+    {
+        _img.enabled = true;
+        _img.sprite = logo;
+    }
+    public void DeactivateHud()
+    {
+        _img.enabled = false;
+    }
+}
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Hud/Hud.cs:24]: add line for EOF

File: Assets/Astralization/Scripts/Interactables/Interactable.cs [hunk 2]
```diff
@@ -23,6 +23,9 @@ public abstract class Interactable : MonoBehaviour, IInteractable
     [Tooltip("The Game Object for the icon")]
     public GameObject InteractableIcon;
 
+    [Tooltip("The Sprite for the logo")]
+    public Sprite InteractableLogo;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Interactables/Interactable.cs:27]: put this in item class and set it private with SerializeField

File: Assets/Astralization/Scripts/Player/Inventory.cs [hunk 3]
```diff
@@ -177,7 +183,7 @@ private void ChangeActiveItem(int newIdx)
         // Show active item
         _activeItem?.ShowItem();
 
-        // TODO: Change active item display on HUD
+        if (_activeItem) _hud.ActivateHud(_activeItem.InteractableLogo);
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Player/Inventory.cs:186]: add else: deactivate hud

**Review by rafiftaris — CHANGES_REQUESTED** [review]
- rename hud folder to UI
- don't forget to add test 😉

File: Assets/Astralization/Scripts/UI/Hud.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+/*
+ * Class to control HUD.
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/UI/Hud.cs:5]: Class to control Item HUD

File: Assets/Astralization/Scripts/UI/Hud.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ */
+public class Hud : MonoBehaviour
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/UI/Hud.cs:7]: Rename to HudItem

File: Assets/Astralization/Scripts/Player/Inventory.cs [hunk 6]
```diff
@@ -45,6 +45,10 @@ public class Inventory : MonoBehaviour, IInventory
 
     [Tooltip("Overworld Item in Level")]
     public GameObject OverworldItem;
+
+    [Tooltip("Hud for Item in Level")]
+    [SerializeField]
+    private Hud _hud;
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Player/Inventory.cs:51]: rename adjust with class name

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
Canvas object should not be put in a prefab.

Add Canvas object in scene and make the HUD prefab only consist of the script and the childe Placeholder with image component. Also, rename the "Placholder" child to "Logo" and Hud prefab to ItemHud (adjust with class name)

**Review by rafiftaris — CHANGES_REQUESTED** [review]
Changes dikit hehe

**Review by rafiftaris — CHANGES_REQUESTED** [review]
Also test on discard and scroll

**Review by rafiftaris — APPROVED** [review]
nicee

**Review by raihansyahdean — APPROVED** [review]
Well done!

**Pull request merged**